Tasgrid/tasgridWrapper.cpp

```diff
@@ -600,6 +626,38 @@ bool TasgridWrapper::getIntegrate(){
 
     return true;
 }
+bool TasgridWrapper::getDifferentiate(){
+    if (grid.getNumLoaded() == 0){
+        cerr << "ERROR: no values loaded in the grid, cannot evaluate!" << endl;
+        return false;
+    }
+    if (grid.getNumOutputs() == 0){
+        cerr << "ERROR: no outputs set for the grid, nothing to evaluate!" << endl;
+        return false;
+    }
+    auto x = readMatrix(xfilename);
+    if (x.getStride() != (size_t) grid.getNumDimensions()){
+        cerr << "ERROR: grid is set for " << grid.getNumDimensions() << " dimensions, but " << xfilename << " specifies " << x.getStride() << endl;
+        return false;
+    }
+    if (x.empty()){
+        cerr << "ERROR: no points specified in " << xfilename << endl;
+        return false;
+    }
+    int num_in = grid.getNumDimensions();
+    int num_out = grid.getNumOutputs();
+    int num_points = x.getNumStrips();
+    std::vector<double> result(num_points * num_in * num_out);
+    std::vector<double> x_vec = x.release();
```

Comment:
Check `getDiffWeights()` and how much cleaner this becomes when you use `Data2D` as opposed to converting to a regular vector and then using complicated indexing.

Reply:
Changed to use `Data2D` instead of `std::vector<double>`.